src/lib/navigation/Nav.svelte

```diff
@@ -0,0 +1,153 @@
+<script>
+  import MenuToggleBtn from "$lib/input/MenuToggleBtn.svelte";
+  import NavItem from "$lib/navigation/NavItem.svelte";
+
+  let pages = [
+    { title: "Home", ref: "/" },
+    { title: "Over Ons", ref: "/over" },
+    { title: "Events", ref: "/events" },
+    { title: "Publicaties", ref: "/publicaties" },
+    { title: "Leden", ref: "/members" },
+    { title: "Vacatures", ref: "/vacatures" },
+  ];
+
+  let pagesCTA = [
+    { title: "Inloggen", ref: "/inloggen" },
+    { title: "Join", ref: "/join" },
+  ];
+
+  let allPages = [...pages, ...pagesCTA];
+
+  let openVacancies = 0;
+  const delay = 1750;
+  let hasAnimated = false;
+  let isOpen = false;
+
+  function toggle() {
+    isOpen = !isOpen;
+
+    // Trigger the animation only the first time the menu opens
+    if (isOpen && !hasAnimated) {
+      hasAnimated = true; // Mark animation as done
+      startVacancyAnimation();
+    }
+  }
+
+  async function startVacancyAnimation() {
+    try {
+      const response = await fetch(
+        "https://fdnd-agency.directus.app/items/dda_agencies_vacancies/"
+      );
+      const vacatures = await response.json();
+      const totalVacancies = vacatures.data ? vacatures.data.length : 0;
+
+      // Animate the counter to transition from 0 to the fetched vacancy count
+      animateCounter(openVacancies, totalVacancies, delay);
+    } catch (error) {
+      console.error("Error fetching vacancies:", error);
+    }
+  }
+
+  function animateCounter(start, end, duration) {
+    const range = end - start;
+    const startTime = performance.now();
+
+    function easeOut(t) {
+      return t * (2 - t);
+    }
+
+    function update(currentTime) {
+      const elapsedTime = currentTime - startTime;
+      const normalizedTime = Math.min(elapsedTime / duration, 1);
+      const easedProgress = easeOut(normalizedTime); // Apply easing function
+      openVacancies = Math.floor(start + range * easedProgress);
+
+      if (normalizedTime < 1) {
+        requestAnimationFrame(update); // Continue the animation
+      }
+    }
+
+    requestAnimationFrame(update);
+  }
```

Comment:
je zou bij je animatie wat meer comments kunnen toevoegen over wat de code precies doet.
De code is ziet er goed uit het is alleen wel heel erg veel in 1 pull request.

Reply:
Ja sorry, vergeten die vorige apart te doen en deze branch op de ander gebaseerd. Zal er volgende keer op letten! 
Ik zal deze code ook nog iets verduidelijken